salt/tests/unit/modules/test_metalk8s_kubernetes_utils.py

```diff
@@ -0,0 +1,221 @@
+
+import metalk8s_kubernetes_utils
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "files", "test_metalk8s_kubernetes_utils.yaml"
+)
+with open(YAML_TESTS_FILE) as fd:
+    YAML_TESTS_CASES = yaml.safe_load(fd)
+
+
+class Metalk8sKubernetesUtilsTestCase(TestCase, LoaderModuleMockMixin):
+    """
+    TestCase for `metalk8s_kubernetes_utils` module
+    """
+    loader_module = metalk8s_kubernetes_utils
+
+    def test_virtual_success(self):
+        """
+        Tests the return of `__virtual__` function, success
+        """
+        reload(metalk8s_kubernetes_utils)
+        self.assertEqual(
+            metalk8s_kubernetes_utils.__virtual__(),
+            'metalk8s_kubernetes'
+        )
+
+    def test_virtual_fail_import(self):
+        """
+        Tests the return of `__virtual__` function, fail import
+        """
+        with ForceImportErrorOn(
+            "kubernetes.client",
+            "kubernetes.config",
+            ('urllib3.exceptions', 'HTTPError')
+        ):
+            reload(metalk8s_kubernetes_utils)
+            self.assertTupleEqual(
+                metalk8s_kubernetes_utils.__virtual__(),
+                (
+                    False,
+                    "Missing dependencies: kubernetes.client, "
+                    "kubernetes.config, urllib3"
+                )
+            )
+
+    @parameterized.expand([
+        param.explicit(kwargs=test_case)
+        for test_case in YAML_TESTS_CASES["get_kubeconfig"]
+    ])
+    def test_get_kubeconfig(
+        self,
+        result,
+        api_server_pillar=None,
+        config_options=None,
+        kwargs={},
+    ):
+        """
+        Tests the return of `get_kubeconfig` function
+        """
+        pillar_dict = {}
+        if api_server_pillar:
+            pillar_dict = {'metalk8s': {'api_server': api_server_pillar}}
+
+        config_options_dict = {}
+        if config_options:
+            config_options_dict = config_options
+
+        config_options_mock = MagicMock(side_effect=config_options_dict.get)
+
+        salt_dict = {
+            'config.option': config_options_mock,
+        }
+
+        with patch.dict(metalk8s_kubernetes_utils.__pillar__, pillar_dict), \
+                patch.dict(metalk8s_kubernetes_utils.__salt__, salt_dict):
+            self.assertEqual(
+                tuple(result),
+                metalk8s_kubernetes_utils.get_kubeconfig(**kwargs)
+            )
+
+    @parameterized.expand([
+        (['my_kubeconfig.conf', 'my_context'], {"major": "1", "go_version": "go1.13.8"}),
+        (['my_kubeconfig.conf', 'my_context'], 'Failed to get version info', True, True),
+    ])
+    def test_get_version_info(
+        self,
+        kubeconfig,
+        result,
+        error=False,
+        raises=False,
+        **kwargs
+    ):
+        """
+        Tests the return of `get_version_info` function
+        """
+        kubeconfig_mock = MagicMock(return_value=tuple(kubeconfig))
```

Comment:
Just return a random value we don't care about the exact value, hard here

```suggestion
        kubeconfig_mock = MagicMock(return_value=('my_kubeconfig.conf', 'my_context'))
```